executor/chroot/chroot_executor.go

```diff
@@ -27,191 +26,80 @@ func (e *Executor) Configure(workspacePath string) {
 	e.workspacePath = workspacePath
 }
 
-func (x *Executor) Run(formula def.Formula) (job def.Job, outs []def.Output) {
-	try.Do(func() {
-		job, outs = x.run(formula)
-	}).Catch(input.Error, func(e *errors.Error) {
-		// REVIEW: also directly pass input/output system errors up?  or, since we may have to gather several, put them in a group and wrap them in a "prereqs failed" executor error?
-		panic(e)
-	}).Catch(Error, func(e *errors.Error) {
-		panic(e)
-	}).CatchAll(func(err error) {
-		panic(UnknownError.Wrap(err))
-	}).Done()
-	return
-}
+func (e *Executor) Start(f def.Formula) def.Job {
 
-func (x *Executor) run(formula def.Formula) (def.Job, []def.Output) {
 	// Prepare the forumla for execution on this host
-	def.ValidateAll(&formula)
+	def.ValidateAll(&f)
+	job := basicjob.New()
 
-	// make up a job id
-	jobID := def.JobID(guid.New())
+	go func(){
+		// Run the formula in a temporary directory
+		flak.WithDir(func(dir string) {
+			job.Result = e.Run(f, job, dir)
+		}, e.workspacePath, "job", string(job.Id()))
 
-	// make a rootfs in our workspace using the jobID
-	jobPath := filepath.Join(x.workspacePath, "job", string(jobID))
-	rootfsPath := filepath.Join(jobPath, "rootfs")
-	if err := os.MkdirAll(rootfsPath, 0755); err != nil {
-		panic(Error.Wrap(errors.IOError.Wrap(err))) // REVIEW: WorkspaceIOError?  or a flag that indicates "wow, super hosed"?
-	}
-
-	// be ready to cleanup the job's filesystems
-	cleanup := func() {
-		if err := os.RemoveAll(jobPath); err != nil {
-			// Note that since this executor doesn't include PID namespacing, it's altogether easy for a runaway process to still have open FDs.
-			// should probably just log in prod mode (but still blow up when running in a test).
-			panic(Error.Wrap(errors.IOError.Wrap(err)))
-		}
-	}
+		// Directory is clean; job complete
+		close(job.WaitChan)
+	}()
 
-	var job def.Job
-	try.Do(func() {
-		// prep inputs
-		x.prepareInputs(rootfsPath, formula.Inputs)
+	return job
+}
 
-		// prep outputs
-		// TODO implement some outputs!
+// Executes a job, catching any panics.
+func (e *Executor) Run(f def.Formula, j def.Job, d string) def.JobResult {
+	var r def.JobResult
 
-		// sandbox up and invoke the real job
-		// this starts the job and immediately returns with a promise
-		job = x.invokeTask(rootfsPath, formula)
+	// TODO: wrap is bananas, wat do
+	try.Do(func() {
+		r = e.Execute(f, j, d)
+	}).Catch(errors.HierarchicalError, func(err *errors.Error) {
```

Comment:
TODO can go, I think this looks solved below?

`s/errors.HierarchicalError/executor.Error/`.  We can&should limit the breadth of tree of errors this can emit... things coming out of this that aren't `executor.Error` should be considered bugs and can get bundled into `executor.UnknownError`.  Probably makes sense to whitelist `input.Error` and `output.Error` the same way.